lib/vagrant-google/action/read_ssh_info.rb

```diff
@@ -42,11 +47,19 @@ def read_ssh_info(google, machine)
             return nil
           end
 
-          # Read SSH network info
-          return {
-            :host => server.public_ip_address,
-            :port => 22
-          }
+          if use_private_ip then
```

Comment:
THanks for the feedback. This is my first code in ruby that does anything useful. I'll correct that to the flow you suggested: 
1.set ssh_info to use the public_ip
2.override with private_ip if use_private_IP is set